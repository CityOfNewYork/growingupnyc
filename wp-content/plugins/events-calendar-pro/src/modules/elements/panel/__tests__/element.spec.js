import React from 'react';
import renderer from 'react-test-renderer';
import Panel from '../element';

describe( 'Panel', () => {
	test( 'should match snapshot', () => {
		const component = renderer.create(
			<Panel
				count={ 2 }
				isExpanded={ false }
				onHeaderClick={ jest.fn() }
				panelTitle="Panel title"
			>
				<span>Test children</span>
			</Panel>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should display children', () => {
		const component = renderer.create(
			<Panel
				count={ 2 }
				isExpanded={ true }
				onHeaderClick={ jest.fn() }
				panelTitle="Panel title"
			>
				<span>Test children</span>
			</Panel>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
