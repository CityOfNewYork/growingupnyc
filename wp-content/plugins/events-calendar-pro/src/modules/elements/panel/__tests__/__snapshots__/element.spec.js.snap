// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should display children 1`] = `
<section>
  <header
    className="tribe-editor__events-pro__panel-header tribe-editor__events-pro__panel-header--expanded"
  >
    <button
      className="tribe-editor__events-pro__panel-header-button"
      onClick={[MockFunction]}
      type="button"
    >
      <span
        className="tribe-editor__events-pro__panel-header-button-text"
      >
        <span
          className="tribe-editor__events-pro__panel-header-button-title"
        >
          Panel title
        </span>
      </span>
      <<svg>icon mock</svg> />
    </button>
  </header>
  <span>
    Test children
  </span>
</section>
`;

exports[`Panel should match snapshot 1`] = `
<section>
  <header
    className="tribe-editor__events-pro__panel-header"
  >
    <button
      className="tribe-editor__events-pro__panel-header-button"
      onClick={[MockFunction]}
      type="button"
    >
      <span
        className="tribe-editor__events-pro__panel-header-button-text"
      >
        <span
          className="tribe-editor__events-pro__panel-header-button-title"
        >
          Panel title
        </span>
        <span
          className="tribe-editor__events-pro__panel-header-button-count"
        >
          (2)
        </span>
      </span>
      <<svg>icon mock</svg> />
    </button>
  </header>
</section>
`;
